sdk/generic-sdk-core/generic-core-http-okhttp/src/main/java/com/generic/core/http/okhttp/OkHttpHttpClient.java

```diff
@@ -138,33 +138,36 @@ private static long getRequestContentLength(BinaryData content, Headers headers)
         return contentLength;
     }
 
-    private static Response<?> toResponse(HttpRequest request, okhttp3.Response response, boolean eagerlyReadResponse,
-                                          boolean ignoreResponseBody, boolean eagerlyConvertHeaders) throws IOException {
-        /*// For now, eagerlyReadResponse and ignoreResponseBody works the same.
-        if (ignoreResponseBody) {
-            ResponseBody body = response.body();
-
-            if (body != null) {
-                if (body.contentLength() > 0) {
-                    LOGGER.log(LogLevel.WARNING, () -> "Received HTTP response body when one wasn't expected. "
-                        + "Response body will be ignored as directed.");
-                }
+    private Response<?> toResponse(HttpRequest request, okhttp3.Response response, boolean eagerlyReadResponse,
+                                   boolean ignoreResponseBody, boolean eagerlyConvertHeaders) throws IOException {
+        okhttp3.Headers responseHeaders = response.headers();
 
-                body.close();
+        if (isTextEventStream(responseHeaders) && response.body() != null) {
+            ServerSentEventListener listener = request.getServerSentEventListener();
+            if (listener != null) {
+                processTextEventStream(request, httpRequestConsumer -> this.send(request), response.body().byteStream(), listener, LOGGER);
+            } else {
+                throw LOGGER.logThrowableAsError(new RuntimeException("No listener attached to the request."));
```

Comment:
nit: The exception message can be more specific - "No ServerSentEventListener attached to HttpRequest to handle the text/event-stream response". This exception is likely due to a bug in client library.